pkg/registry/core/serviceaccount/storage/token.go

```diff
@@ -89,6 +90,11 @@ func (r *TokenREST) Create(ctx context.Context, name string, obj runtime.Object,
 		gvk := schema.FromAPIVersionAndKind(ref.APIVersion, ref.Kind)
 		switch {
 		case gvk.Group == "" && gvk.Kind == "Pod":
+			if r.pods == nil {
+				// TODO(sttts): kubelet will try to create bound tokens against the pcluster, but those obviously
+				//              will not work against kcp. ¯\_(ツ)_/¯
+				return nil, errors.NewBadRequest("spec.BoundObjectRef.Kind equal to Pod is invalid")
```

Comment:
nit: unrelated change

Reply:
moved into its own commit

Reply:
For future, I suggest these kinds of changes be submitted as separate PRs rather than separate commits in a larger PR with unrelated changes. Ideally a PR has changes that are logically consistent to ensure review can focus on the important details.